quickstart/backend/src/main/java/com/digitalasset/quickstart/service/AppInstallRequestsApiImpl.java

```diff
@@ -42,128 +51,199 @@ public AppInstallRequestsApiImpl(
+                        })
+                );
     }
 
     @Override
+    @WithSpan
     public CompletableFuture<ResponseEntity<Void>> rejectAppInstallRequest(
-            String contractId,
-            String commandId,
+            @SpanAttribute("appInstall.contractId") String contractId,
+            @SpanAttribute("appInstall.commandId") String commandId,
             AppInstallRequestReject appInstallRequestReject
     ) {
-        logger.info("Rejecting AppInstallRequest with contractId: {}", contractId);
-        String providerParty = authenticatedPartyService.getPartyOrFail();
-        return damlRepository.findAppInstallRequestById(contractId).thenCompose(contract -> {
-            // Create choice instance
-            quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Reject choice =
-                    new quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Reject(
-                            new quickstart_licensing.licensing.util.Metadata(appInstallRequestReject.getMeta().getData())
-                    );
-
-            return ledger.exerciseAndGetResult(providerParty, contract.contractId, choice, commandId)
-                    .thenApply(result -> ResponseEntity.ok().<Void>build())
-                    .exceptionally(ex -> {
-                        logger.error("Error rejecting AppInstallRequest", ex);
-                        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-                    });
-        }).exceptionally(ex -> {
-            logger.error("Error fetching AppInstallRequest with contractId: {}", contractId, ex);
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-        });
+        Span methodSpan = Span.current();
+        Context parentContext = Context.current();
+
+        methodSpan.addEvent("Starting rejectAppInstallRequest");
+        methodSpan.setAttribute("contractId", contractId);
+        methodSpan.setAttribute("commandId", commandId);
+        logger.info("rejectAppInstallRequest: contractId='{}', commandId='{}'", contractId, commandId);
+
+        return authenticatedPartyService.getPartyOrFail()
+                .thenCompose(providerParty ->
+                        damlRepository.findAppInstallRequestById(contractId)
+                                .thenCompose(contract -> {
+                                    methodSpan.addEvent("Fetched contract, exercising AppInstallRequest_Reject choice");
+
+                                    quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Reject choice =
+                                            new quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Reject(
+                                                    new quickstart_licensing.licensing.util.Metadata(appInstallRequestReject.getMeta().getData())
+                                            );
+
```

Comment:
This is the business to ledger transition point, where we should be logging both the business context and the relevant correlation ids to link this to LedgerApi logs.